clients/metadata/device.go

```diff
@@ -22,69 +22,69 @@ import (
 
 	"github.com/edgexfoundry/go-mod-core-contracts/clients"
 	"github.com/edgexfoundry/go-mod-core-contracts/clients/interfaces"
-	"github.com/edgexfoundry/go-mod-core-contracts/clients/types"
-	"github.com/edgexfoundry/go-mod-core-contracts/clients/urlclient"
 	"github.com/edgexfoundry/go-mod-core-contracts/models"
 )
 
 // DeviceClient defines the interface for interactions with the Device endpoint on core-metadata.
 type DeviceClient interface {
 	// Add creates a new device
-	Add(dev *models.Device, ctx context.Context) (string, error)
+	Add(ctx context.Context, dev *models.Device) (string, error)
 	// Delete eliminates a device for the specified ID
-	Delete(id string, ctx context.Context) error
+	Delete(ctx context.Context, id string) error
 	// DeleteByName eliminates a device for the specified name
 	DeleteByName(name string, ctx context.Context) error
```

Comment:
context.Context should be first parameter.

Please review PR and correct all instances.